src/main/java/mekanism/common/tile/machine/TileEntityDigitalMiner.java

```diff
@@ -441,10 +442,12 @@ private boolean canMine(BlockPos pos) {
             return false;
         }
         BlockState state = world.getBlockState(pos);
-        PlayerEntity dummy = Objects.requireNonNull(Mekanism.proxy.getDummyPlayer((ServerWorld) world, getPos()).get());
-        BlockEvent.BreakEvent event = new BlockEvent.BreakEvent(world, pos, state, dummy);
-        MinecraftForge.EVENT_BUS.post(event);
-        return !event.isCanceled();
+        return MekFakePlayer.withFakePlayer((ServerWorld) world, this.pos.getX(), this.pos.getY(), this.pos.getZ(), dummy -> {
+            dummy.setEmulatingUUID(getSecurity().getOwnerUUID());//pretend to be the owner
+            BlockEvent.BreakEvent event = new BlockEvent.BreakEvent(world, pos, state, dummy);
+            MinecraftForge.EVENT_BUS.post(event);
+            return !event.isCanceled();
```

Comment:
We might as well actually clean this line up because `post` returns if it is canceled or not for cancellable events. So this could be `return !MinecraftForge.EVENT_BUS.post(new BlockEvent.BreakEvent(world, pos, state, dummy));` (It might have to be inverted I don't remember for certain if it returns true for if it is cancelled or not, the javadocs on `post` should say though)